src/openforms/utils/json_logic.py

```diff
@@ -49,3 +52,174 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
         return JsonLogicTest.from_expression(value)
 
     raise NotImplementedError(f"Unknown value type: {type(value)}")
+
+
+@dataclass
+class ComponentMeta:
+    form_step: "FormStep"
+    component: Dict[str, Any]
+
+
+@dataclass
+class IntrospectionResult:
+
+    expression: dict
+    rule_tree: Dict[Any, str]
+
+    def as_string(self) -> str:
+        return stringify_json_tree(self.rule_tree)
+
+    def get_input_components(self):
+        return get_leaves(self.rule_tree)
+
+
+def introspect_json_logic(
+    expression: Dict[Any, str],
+    components_map: Dict[Any, ComponentMeta],
+    input_data: Dict[str, Any],
+) -> IntrospectionResult:
+    logic_expression = JsonLogicTest.from_expression(expression)
+    rule_tree = convert_json_logic_in_json_tree(
+        logic_expression, components_map, input_data
+    )
+    print(rule_tree)
+    return IntrospectionResult(expression=expression, rule_tree=rule_tree)
+
+
+def find_json_logic_test_value(value: JsonLogicTest) -> str:
+    """returns the value inside a JsonLogicTest"""
+    if isinstance(value, JsonLogicTest) and value.operator in ["date", "var"]:
+        return find_json_logic_test_value(value.values[0])
+    else:
+        return value
+
+
+def convert_json_logic_in_json_tree(logics, component_map, resulting_data):
+    """returns a tree of nodes which are representing either an operation or a component
+
+    This function transforms a logic_rule into a tree with informations related to the
+    components involved in the rule such as its `value`, `label`, `step name`...
+
+    :param logics:  :class:`JsonLogicTest` the logic rule
+    :param keys_mapped_to_component: dict of component with component key as key
+    :param keys_mapped_to_form: dict of form with component key as key
+    :param resulting_data: actual data from the submission form
+
+    Example of data structure returned:
+
+    {
+        "nodeType":"operation",
+        "operator":">",
+        "children":[
+            {
+                "nodeType":"leaf",
+                "key":"birthdate",
+                "actualValue":"2022-06-21",
+                "stepName":"definition-birthdate",
+                "label":"Birthdate"
+            }
+        ],
+        "value":"2022-06-20"
+    }
+
+    :rtype: dict
+    :return: rule with additional informations related to the involved component
```

Comment:
this *would* also benefit from dataclasses rather than dicts (helps a lot with the type hinting & readability)

Reply:
Something like JsonLogicOperation and JsonLogicValue ?